MapboxNavigation/CarPlayNavigationViewController.swift

```diff
@@ -420,6 +440,23 @@ public class CarPlayNavigationViewController: UIViewController {
     }
 }
 
+@available(iOS 12.0, *)
+extension CarPlayNavigationViewController: NavigationMapViewDelegate {
+    public func navigationMapViewUserAnchorPoint(_ mapView: NavigationMapView) -> CGPoint {
+        // Inset by the content inset to avoid application-defined content.
+        var contentFrame = UIEdgeInsetsInsetRect(mapView.bounds, mapView.contentInset)
+        
+        // Avoid letting the puck go partially off-screen, and add a comfortable padding beyond that.
+        let courseViewBounds = mapView.userCourseView?.bounds ?? .zero
+        contentFrame = contentFrame.insetBy(dx: min(50 + courseViewBounds.width / 2.0, contentFrame.width / 2.0),
+                                            dy: min(50 + courseViewBounds.height / 2.0, contentFrame.height / 2.0))
```

Comment:
nit: probably a variable for these magic numbers will aid self-documentation here.

Reply:
Indeed, these couple lines were copied from `NavigationMapView.userAnchorPoint` to avoid impacting non-CarPlay maps. While the logic probably doesn’t warrant a separate function, I agree that the additional padding should be centralized in a constant.